### ISSUE TITLE
Associatedtype length warning

### ISSUE THREAD

**Issue by yurevich1** [realm/SwiftLint#2025]
### Bug Report
I not sure how can I disable this warning. Is it a bug?

import Foundation

```
protocol JSONParser {
    associatedtype T
    /// Generic parser
    func parse(data: Data) -> T?
}
```

at line `associatedtype T` 

> Type Name Violation: Type name should be between 3 and 60 characters long: 'T' (type_name)

### Environment

* SwiftLint version0.24.2?
* brew
* Paste your configuration file:

```
disabled_rules:
- force_cast
excluded: # paths to ignore during linting. overridden by `included`.
  - Carthage
  - Pods
identifier_name:
 max_length: # warning and error
    warning: 60
    error: 100
 excluded:
    - id
type_name:
 min_length: 3 # only warning
 max_length: # warning and error
    warning: 60
    error: 100
file_length:
 warning: 1000
 error: 2000
 excluded:
    - T
```

**Comment by SDGGiesbrecht** [comment]
@yurevich1 

Did you mean to have...
```yaml
excluded:
  - T
```
...under `type_name` instead of `file_length`?

**Comment by yurevich1** [comment]
I found the solution. It was an xcode bug. After restarting this feature works as expected.
